ci/jenkins/test-rancher.sh

```diff
@@ -0,0 +1,387 @@
+}
+
+function clean_vm_agent {
+    echo "====== Cleanup Antrea-Agent Installation on the VM ======"
+    echo "Revert to snapshot defaultState"
+    kubectl get nodes -o wide --no-headers=true | awk '{print $1}' | while read HOSTNAME; do
+        govc snapshot.revert -k=true -vm=${HOSTNAME} defaultState
+    done
+}
+
+function clean_up_one_ns {
+    ns=$1
+    kubectl get pod -n "${ns}" --no-headers=true | awk '{print $1}' | while read pod_name; do
+        kubectl delete pod "${pod_name}" -n "${ns}" --force --grace-period 0
+    done
+    kubectl delete ns "${ns}" --ignore-not-found=true || true
+}
+
+function deliver_antrea {
+    echo "====== Cleanup Antrea Installation ======"
+    clean_up_one_ns "monitoring" || true
+    clean_up_one_ns "antrea-ipam-test-11" || true
+    clean_up_one_ns "antrea-ipam-test-12" || true
+    clean_up_one_ns "antrea-ipam-test" || true
+    clean_up_one_ns "antrea-test" || true
+    kubectl delete -f ${WORKDIR}/antrea-prometheus.yml || true
+    kubectl delete daemonset antrea-agent -n kube-system || true
+    kubectl delete -f ${WORKDIR}/antrea.yml || true
+    if [[ $TESTBED_TYPE == "flexible-ipam" ]]; then
+        redeploy_k8s_if_ip_mode_changes
+    fi
+
+    echo "====== Building Antrea for the Following Commit ======"
+    export GO111MODULE=on
+    export GOPATH=${WORKDIR}/go
+    export GOROOT=/usr/local/go
+    export GOCACHE="${WORKSPACE}/../gocache"
+    export PATH=${GOROOT}/bin:$PATH
+
+    git show --numstat
+    make clean
+    ${CLEAN_STALE_IMAGES}
+    if [[ ! "${TESTCASE}" =~ "e2e" && "${DOCKER_REGISTRY}" != "" ]]; then
+        docker pull "${DOCKER_REGISTRY}/antrea/sonobuoy-systemd-logs:v0.3"
+        docker tag "${DOCKER_REGISTRY}/antrea/sonobuoy-systemd-logs:v0.3" "sonobuoy/systemd-logs:v0.3"
+    fi
+    chmod -R g-w build/images/ovs
+    chmod -R g-w build/images/base
+    DOCKER_REGISTRY="${DOCKER_REGISTRY}" ./hack/build-antrea-linux-all.sh --pull
+    make flow-aggregator-image
+
+    # Enable verbose log for troubleshooting.
+    sed -i "s/--v=0/--v=4/g" build/yamls/antrea.yml
+
+    echo "=== Fill serviceCIDRv6 and serviceCIDR ==="
+    # It is unnecessary for cluster with AntreaProxy enabled.
+    SVCCIDRS=$(kubectl cluster-info dump | grep service-cluster-ip-range | head -n 1 | cut -d'=' -f2 | cut -d'"' -f1)
+    echo "Service CIDRs are $SVCCIDRS"
+    regexV6="^([0-9a-fA-F]{0,4}:){1,7}[0-9a-fA-F]{0,4}.*$"
+    IFS=',' read -ra CIDRS <<< "$SVCCIDRS"
+    for cidr in "${CIDRS[@]}"; do
+        if [[ ${cidr} =~ ${regexV6} ]]; then
+            sed -i "s|#serviceCIDRv6:|serviceCIDRv6: ${cidr}|g" build/yamls/antrea.yml
+        else
+            sed -i "s|#serviceCIDR: 10.96.0.0/12|serviceCIDR: ${cidr}|g" build/yamls/antrea.yml
+        fi
+    done
+
+    echo  "=== Append antrea-prometheus.yml to antrea.yml ==="
+    echo "---" >> build/yamls/antrea.yml
+    cat build/yamls/antrea-prometheus.yml >> build/yamls/antrea.yml
+
+    if [[ $TESTBED_TYPE == "flexible-ipam" ]]; then
```

Comment:
The type processing code should be refactored.

Reply:
okay